Tests/Api/Testably.Abstractions.Api.Tests/Expected/Testably.Abstractions.Interface_net8.0.txt

```diff
@@ -0,0 +1,214 @@
+        public virtual System.ReadOnlySpan<char> GetFileNameWithoutExtension(System.ReadOnlySpan<char> path) { }
+        [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNull("path")]
+        public virtual string? GetFileNameWithoutExtension(string? path) { }
+        public virtual string GetFullPath(string path) { }
+        public virtual string GetFullPath(string path, string basePath) { }
+        public virtual char[] GetInvalidFileNameChars() { }
+        public virtual char[] GetInvalidPathChars() { }
+        public virtual System.ReadOnlySpan<char> GetPathRoot(System.ReadOnlySpan<char> path) { }
+        public virtual string? GetPathRoot(string? path) { }
+        public virtual string GetRandomFileName() { }
+        public virtual string GetRelativePath(string relativeTo, string path) { }
+        [System.Obsolete("Insecure temporary file creation methods should not be used. Use `Path.Combine(Pa" +
+            "th.GetTempPath(), Path.GetRandomFileName())` instead.")]
+        public virtual string GetTempFileName() { }
+        public virtual string GetTempPath() { }
+        public virtual bool HasExtension(System.ReadOnlySpan<char> path) { }
+        public virtual bool HasExtension([System.Diagnostics.CodeAnalysis.NotNullWhen(true)] string? path) { }
+        public virtual bool IsPathFullyQualified(System.ReadOnlySpan<char> path) { }
+        public virtual bool IsPathFullyQualified(string path) { }
+        public virtual bool IsPathRooted(System.ReadOnlySpan<char> path) { }
+        public virtual bool IsPathRooted(string? path) { }
+        public virtual string Join(params string?[] paths) { }
+        public virtual string Join(System.ReadOnlySpan<char> path1, System.ReadOnlySpan<char> path2) { }
+        public virtual string Join(string? path1, string? path2) { }
+        public virtual string Join(System.ReadOnlySpan<char> path1, System.ReadOnlySpan<char> path2, System.ReadOnlySpan<char> path3) { }
+        public virtual string Join(string? path1, string? path2, string? path3) { }
+        public virtual string Join(System.ReadOnlySpan<char> path1, System.ReadOnlySpan<char> path2, System.ReadOnlySpan<char> path3, System.ReadOnlySpan<char> path4) { }
+        public virtual string Join(string? path1, string? path2, string? path3, string? path4) { }
+        public virtual System.ReadOnlySpan<char> TrimEndingDirectorySeparator(System.ReadOnlySpan<char> path) { }
+        public virtual string TrimEndingDirectorySeparator(string path) { }
+        public virtual bool TryJoin(System.ReadOnlySpan<char> path1, System.ReadOnlySpan<char> path2, System.Span<char> destination, out int charsWritten) { }
+        public virtual bool TryJoin(System.ReadOnlySpan<char> path1, System.ReadOnlySpan<char> path2, System.ReadOnlySpan<char> path3, System.Span<char> destination, out int charsWritten) { }
+    }
+    public sealed class RandomWrapper : Testably.Abstractions.RandomSystem.IRandom
```

Comment:
Should this really be public?

Reply:
Yes, it needs to be, otherwise we would have to duplicate the code in the `Testably.Abstractions` and `Testably.Abstractions.Testing` projects or add a reference from one to the other.